### PR TITLE
Support go 1.22 and 1.23 build to fix CVE

### PR DESCRIPTION
**Pull request by smoshiur1237**
This PR will address a CVE, reported in #1354

### CODE REVIEW

**Comment by smoshiur1237** [comment]
@juliusmh I can add support for 1.23 .

**Comment by smoshiur1237** [comment]
@juliusmh I have added support for both 1.22 and 1.23.

**Comment by smoshiur1237** [comment]
@dougbtv @s1061123 @juliusmh  would you please check or review to prioritize this patch and create a new release. The fix for CVE is critical and would appreciate if you please take it in.

**Comment by juliusmh** [comment]
@smoshiur1237 I have no authority in this repository.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ jobs:
   build:
     strategy:
       matrix:
-        go-version: [1.20.x, 1.21.x]
+        go-version: [1.20.x, 1.21.x, 1.22.x, 1.23.x]
```
Comment (s1061123) [inline .github/workflows/build.yml:7]: Hi, can you remove 1.20.x and 1.21.x? I suppose two go version is enough.

**Comment by s1061123** [comment]
Could you please care this one?

https://github.com/k8snetworkplumbingwg/multus-cni/blob/781ecdaecd819c24be9eb2d5e0e45f3dc6d9d358/.github/workflows/release.yml#L18

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.20.x, 1.21.x]
+        go-version: [1.20.x, 1.21.x, 1.22.x, 1.23.x]
```
Comment (smoshiur1237) [inline .github/workflows/test.yml:7]: @s1061123  do I need to remove older version and keep same as other file to keep two go versions only

**Comment by smoshiur1237** [comment]
> Could you please care this one?
> 
> https://github.com/k8snetworkplumbingwg/multus-cni/blob/781ecdaecd819c24be9eb2d5e0e45f3dc6d9d358/.github/workflows/release.yml#L18

do I change it to 1.22 or 1.23?

**Comment by smoshiur1237** [comment]
@dougbtv @s1061123  Thanks for your review. I have addressed them and pushed the changes. Please let me know if it is ok?

**Comment by smoshiur1237** [comment]
> > Could you please care this one?
> > https://github.com/k8snetworkplumbingwg/multus-cni/blob/781ecdaecd819c24be9eb2d5e0e45f3dc6d9d358/.github/workflows/release.yml#L18
> 
> do I change it to 1.22 or 1.23?

I asked @dougbtv and he agrees to keep it for 1.22 as minimum build support stays for 1.22

**Comment by smoshiur1237** [comment]
Thanks @dougbtv and @s1061123  for review and take this in. would you please tag it as v4.1.4 release?

**Pull request merged**